### ISSUE TITLE
Gentle auto-import that respects Scala CLI's directives

### ISSUE THREAD

**Issue by windymelt** [scalameta/metals#7071]
### Is your feature request related to a problem? Please describe.

When I auto-import by completion in Scala Script for Scala CLI, Metals always adds import line top of my file. However, I want to keep Scala CLI directives top of the file...

e.g.

Before auto-import:

```scala
//> using scala 3.6.2

val d = FiniteDuration/* press TAB to complete! */
```

After auto-import:

```scala
import scala.concurrent.duration.FiniteDuration // aww〜〜
//> using scala 3.6.2

val d = FiniteDuration
```

### Describe the solution you'd like

- Metals recognizes Scala CLI directives.
- Metals places imports below directives when auto-import.

### Describe alternatives you've considered

Cut and paste

### Additional context

I'm using Metals 1.4.1 on Emacs.

### Search terms

autoimport  auto-import import

**Comment by kasiaMarek** [comment]
Hi, I can't really reproduce your issue. It puts the import in the correct place for me.

https://github.com/user-attachments/assets/aeaff606-2075-454f-9839-e5d7b8b75523

Are you sure that script is correctly recognised by Metals as Scala CLI script? Meaning, either `scala-cli` is the build server used in your project or the script is imported as a Scala CLI script in Metals.

**Comment by dos65** [comment]
I've seen such issue when there were no `\n` after using directive but forgot to report it

```scala
//> using scala 3.6
val d = FiniteDuration
```

**Comment by kasiaMarek** [comment]
@dos65, it also seems to work for me without the `\n`.

**Comment by dos65** [comment]
@kasiaMarek yep, it works well on scala 3.6. my bad -  the issue was on 2.13

Try this:
```
//> using scala 2.13
object X {
   FiniteDuration@@
}
```

**Comment by kubukoz** [comment]
I see directives and auto-imports keep fighting each other 😆 #4939 #6027
